src/modules/trade/actions/update-trades-in-progress.js

```diff
@@ -91,51 +92,61 @@ export function updateTradesInProgress(marketID, outcomeID, side, numShares, lim
+    if (market.marketType !== SCALAR && limitPrice) {
       cleanLimitPrice = bignumLimit.abs().toFixed() || outcomeTradeInProgress.limitPrice || topOrderPrice
     }
 
+    // if market order limitPrice should be null
+    if (orderType === MARKET) {
+      cleanLimitPrice = null
+    }
+
     const newTradeDetails = {
       side: cleanSide,
       numShares: cleanNumShares === '0' ? undefined : cleanNumShares,
       limitPrice: cleanLimitPrice,
       totalFee: '0',
       totalCost: '0'
     }
-
     // trade actions
-    if (newTradeDetails.side && newTradeDetails.numShares && loginAccount.address) {
+    if (newTradeDetails.side && loginAccount.address && newTradeDetails.numShares) {
       dispatch(loadAccountPositions({ market: marketID }, (err, accountPositions) => {
         if (err) {
           return dispatch({
             type: UPDATE_TRADE_IN_PROGRESS,
             data: { marketID, outcomeID, details: newTradeDetails }
           })
         }
+        const cleanAccountPositions = []
+        for (let i = 0; i < market.numOutcomes; i++) {
+          if (accountPositions[i] && !isNaN(accountPositions[i].numShares)) {
+            cleanAccountPositions.push(accountPositions[i].numShares)
+          } else {
+            cleanAccountPositions.push(0)
+          }
+        }
         const simulatedTrade = augur.trading.simulateTrade({
           orderType: newTradeDetails.side === BUY ? 0 : 1,
           outcome: parseInt(outcomeID, 10),
-          shareBalances: accountPositions,
-          tokenBalance: (loginAccount.ethTokens && loginAccount.ethTokens.toString()) || '0',
+          shareBalances: cleanAccountPositions,
+          tokenBalance: (loginAccount.eth && loginAccount.eth.toString()) || '0',
           userAddress: loginAccount.address,
           minPrice: market.minPrice,
           maxPrice: market.maxPrice,
           price: newTradeDetails.limitPrice,
           shares: newTradeDetails.numShares,
           marketCreatorFeeRate: market.settlementFee,
-          singleOutcomeOrderBook: (orderBooks && orderBooks[marketID]) || {},
+          singleOutcomeOrderBook: (orderBooks && orderBooks[marketID] && orderBooks[marketID][outcomeID]) || {},
           shouldCollectReportingFees: !market.isDisowned,
           reportingFeeRate: market.reportingFeeRate
         })
-        // console.log('simulated trade:', JSON.stringify(simulatedTrade, null, 2));
         const totalFee = new BigNumber(simulatedTrade.settlementFees, 10).plus(new BigNumber(simulatedTrade.gasFees, 10))
         newTradeDetails.totalFee = totalFee.toFixed()
-        newTradeDetails.totalCost = new BigNumber(simulatedTrade.tokensDepleted, 10).neg().toFixed()
+        newTradeDetails.totalCost = simulatedTrade.tokensDepleted
         newTradeDetails.feePercent = totalFee.dividedBy(new BigNumber(simulatedTrade.tokensDepleted, 10)).toFixed()
         if (isNaN(newTradeDetails.feePercent)) newTradeDetails.feePercent = '0'
```

Comment:
Under what circumstances is `newTradeDetails.feePercent` a `NaN`?  Is this something that has come up?

Reply:
Not sure, this was there prior to anything i came to change, maybe it's just a sanity check to make sure we can display later in a valid format? I'll keep an eye out for feePercent coming back as NaN